shared/src/Datadog.Trace.ClrProfiler.Native/dynamic_dispatcher.cpp

```diff
@@ -22,10 +22,43 @@ namespace datadog::shared::nativeloader
     DynamicDispatcherImpl::DynamicDispatcherImpl() :
         m_continuousProfilerInstance(nullptr),
         m_tracerInstance(nullptr),
-        m_customInstance(nullptr)
+        m_customInstance(nullptr),
+        m_initialized(false),
+        m_initializationResult(E_NOT_SET)
     {
     }
 
+    HRESULT DynamicDispatcherImpl::Initialize()
+    {
+        if (m_initialized)
+        {
+            return m_initializationResult;
+        }
+
+        m_initialized = true;
+
+        LoadConfiguration(GetConfigurationFilePath());
+
+        m_initializationResult = LoadClassFactory(IID_IClassFactory);
```

Comment:
I assume it's irrelevant/can't happen in practice, but should `m_initialized = true` be after the first `m_initializationResult =`?

Reply:
The problem is that I don't want to run `LoadConfiguration` multiple times if `Initialize` is called again. Since `m_initializationResult`'s initial value is an error code, I don't think that's a problem.